src/coreComponents/mesh/generators/VTKUtilities.hpp

```diff
@@ -0,0 +1,576 @@
+ * @param vtkArray The source.
+ * @param wrapper The destination.
+ */
+void importRegularField( std::vector< vtkIdType > const & cellIds,
+                         vtkDataArray * vtkArray,
+                         WrapperBase & wrapper );
+
+/**
+ * @brief Print statistics for a vtk mesh
+ *
+ * @param[in] mesh an input mesh
+ * @param cellMap a map of cell lists grouped by type
+ * @param comm the MPI communicator
+ */
+void printMeshStatistics( vtkDataSet & mesh,
+                          CellMapType const & cellMap,
+                          MPI_Comm const comm );
+
+/**
+ * @brief Collect the data to be imported.
+ * @param[in] mesh an input mesh
+ * @param[in] srcFieldNames an array of field names
+ * @return A list of pointers to VTK data arrays.
+ */
+std::vector< vtkDataArray * >
+findArraysForImport( vtkDataSet & mesh,
+                     arrayView1d< string const > const & srcFieldNames );
+
+
+/**
+ * @brief Build the Elements to Nodes indexing
+ *
+ * @tparam INDEX_TYPE the type of the indexes
+ * @tparam POLICY The execution policy
+ * @param mesh an input mesh
+ * @param cells an array of the cells
+ * @return An Elements to Nodes indexing
+ */
+template< typename INDEX_TYPE, typename POLICY >
+ArrayOfArrays< INDEX_TYPE, INDEX_TYPE >
+buildElemToNodesImpl( vtkDataSet & mesh,
+                      vtkSmartPointer< vtkCellArray > const & cells )
+{
+  localIndex const numCells = LvArray::integerConversion< localIndex >( mesh.GetNumberOfCells() );
+  array1d< INDEX_TYPE > nodeCounts( numCells );
+
+  // GetCellSize() is always thread-safe, can run in parallel
+  forAll< parallelHostPolicy >( numCells, [nodeCounts = nodeCounts.toView(), &cells] ( localIndex const cellIdx )
+  {
+    nodeCounts[cellIdx] = LvArray::integerConversion< INDEX_TYPE >( cells->GetCellSize( cellIdx ) );
+  } );
+
+  ArrayOfArrays< INDEX_TYPE, INDEX_TYPE > elemToNodes;
+  elemToNodes.template resizeFromCapacities< parallelHostPolicy >( numCells, nodeCounts.data() );
+
+  vtkIdTypeArray const & globalPointId = *vtkIdTypeArray::FastDownCast( mesh.GetPointData()->GetGlobalIds() );
+
+  // GetCellAtId() is conditionally thread-safe, use POLICY argument
+  forAll< POLICY >( numCells, [&cells, &globalPointId, elemToNodes = elemToNodes.toView()] ( localIndex const cellIdx )
+  {
+    vtkIdType numPts;
+    vtkIdType const * points;
+    cells->GetCellAtId( cellIdx, numPts, points );
+    for( int a = 0; a < numPts; ++a )
+    {
+      vtkIdType const pointIdx = globalPointId.GetValue( points[a] );
+      elemToNodes.emplaceBack( cellIdx, LvArray::integerConversion< INDEX_TYPE >( pointIdx ) );
+    }
+  } );
+
+  return elemToNodes;
+}
+
+/**
+ * @brief Build the Elements to Nodes indexing depending on the execution policy
+ *
+ * @tparam INDEX_TYPE the type of the indexes
+ * @param mesh an input mesh
+ * @return An Elements to Nodes indexing
+ */
+template< typename INDEX_TYPE >
+ArrayOfArrays< INDEX_TYPE, INDEX_TYPE >
+buildElemToNodes( vtkDataSet & mesh )
```

Comment:
Assuming you won't need to call this function directly, outside of `redistributeByCellGraph`, could you move it entirely (definition and declaration, as well as the above `Impl` function) into the `.cpp`?

Reply:
yes